### PR TITLE
Improve Erased Role

### PR DESCRIPTION
**Pull request by Tommy-XL**
Some role erasure fixes and improvements

Now the code itself determines the RoleType of the player and changes it to the desired vanilla role
If the role be works fine, then this role will cease to be experimental

### CODE REVIEW

**Comment by Loonie-Toons** [comment]
How does it determine imp basis crew roles?

File: Roles/Impostor/Eraser.cs [hunk 1]
```diff
@@ -101,7 +100,7 @@ public static void AfterMeetingTasks()
         {
             var player = Utils.GetPlayerById(pc);
             if (player == null) continue;
-            player.RpcSetCustomRole(CustomRolesHelper.GetErasedRole(player.GetCustomRole()));
+            player.RpcSetCustomRole(CustomRolesHelper.GetErasedRole(player.GetCustomRole().GetRoleTypes(), player.GetCustomRole()));
```
Comment (Tommy-XL) [inline Roles/Impostor/Eraser.cs:103]: The code produces ### GetRoleTypes(), accordingly returns the player’s ### RoleTypes and, depending on the role type, returns a vanilla role (naturally, it cannot erase neutral roles)

File: Modules/CustomRolesHelper.cs [hunk 2]
```diff
@@ -145,56 +145,19 @@ static class CustomRolesHelper
...
-                CustomRoles.Guardian => CustomRoles.CrewmateTOHE,
-                CustomRoles.Addict => CustomRoles.EngineerTOHE,
-                CustomRoles.Oracle => CustomRoles.CrewmateTOHE,
-                CustomRoles.Chameleon => CustomRoles.EngineerTOHE,
-                _ => role.IsImpostor() ? CustomRoles.ImpostorTOHE : CustomRoles.CrewmateTOHE,
+                RoleTypes.Crewmate => CustomRoles.CrewmateTOHE,
+                RoleTypes.Scientist => CustomRoles.ScientistTOHE,
+                RoleTypes.Engineer => CustomRoles.EngineerTOHE,
+                RoleTypes.Impostor => CustomRoles.ImpostorTOHE,
+                RoleTypes.Shapeshifter => CustomRoles.ShapeshifterTOHE,
```
Comment (Tommy-XL) [inline Modules/CustomRolesHelper.cs:159]: This is where the code defines the role type

Comment (Loonie-Toons) [reply Modules/CustomRolesHelper.cs:159]: So what if, for example, Sheriff got erased?
According to the host, their role type is Scientist.
For everyone else, it's Impostor.

Comment (Tommy-XL) [reply Modules/CustomRolesHelper.cs:159]: Hmm, I didn't think this through, I need to make corrections, lol

Comment (Loonie-Toons) [reply Modules/CustomRolesHelper.cs:159]: Yeah, that's why I was asking lol.

**Comment by Tommy-XL** [comment]
Fixed

File: Modules/CustomRolesHelper.cs [hunk 3]
```diff
@@ -883,6 +846,10 @@ CustomRoles.Witness or
             CustomRoles.Monarch or
             CustomRoles.Farseer or
             CustomRoles.SwordsMan or
+            CustomRoles.Admirer or
+            CustomRoles.Divinator or
```
Comment (Loonie-Toons) [inline Modules/CustomRolesHelper.cs:850]: How are Fortune Teller (Divinator) and Merchant taskless? lol

Comment (Tommy-XL) [reply Modules/CustomRolesHelper.cs:850]: They're based on the kill button

Comment (Loonie-Toons) [reply Modules/CustomRolesHelper.cs:850]: They both use a crewmate basis.
Merchant's ability uses tasks, while Fortune Teller uses votes (and has tasks).

Comment (Tommy-XL) [reply Modules/CustomRolesHelper.cs:850]: Hhhh, i will fixed

**Comment by Tommy-XL** [comment]
Done, i hope

**Comment by Loonie-Toons** [comment]
Alright seems good now lol.

**Pull request merged**